source/source_test.go

```diff
@@ -19,6 +20,7 @@ func TestConfigureSource_FailsWhenConfigInvalid(t *testing.T) {
 	con := Source{}
 	err := con.Configure(context.Background(), map[string]string{"foobar": "foobar"})
 
+	// same comments as above
```

Comment:
@WeirdMagician when you get to it, please check the above comments, just so we are sure we got everything covered.

Reply:
Actually it is completed and added to main branch when this PR is created. we didn't tracked it separately and thanks for reminding.
I have used every where errors.is whenever an expected err is already available if not using the string comparing.